R/main.R

```diff
@@ -54,10 +54,37 @@ cursor_to_tibble <- function(cursor, column_names) {
   if (nrow(data) == 0) empty_tibble(column_names) else data
 }
 
-collection <- function(collection_name, connection, column_names = character()) {
+#' Collect table entries
+#'
+#' This collects entries from specific collection and assigns it to reactive value.
+#'
+#' @param collection_name A name of collection.
+#' @param connection Connection object.
+#' @param column_names Names of specific columns.
+#' @param post_process Post process function called before query is run.
+#'
+#' @return Reactive value with connection, collection_name and collection
+#' @export
+#'
+#' @examples
+#'
+#' # Set up a connection to DEFAULT_DB
+#' cn <- connect()
+#' tvs <- collection("tv_shows", cn)
+#' # Take a look at gathered entries
+#' isolate(tvs$collection)
+#'
+#' # Using post_process you might use specific rethinkDB operations
+#' # in rethinker query format before query is run (like filter, orderBy).
+#' # E.g., to gather only tv shows with less than 100 episodes use:
+#' tvs <- collection("tv_shows", cn,
+#'             post_process = function(q) q$filter(function(x) x$bracket("episodes")$lt(100)))
+collection <- function(collection_name, connection, column_names = character(),
+                       post_process = I) {
   make_sure_table_exists(connection, collection_name)
 
-  cursor <- rethinker::r()$db(DEFAULT_DB)$table(collection_name)$run(connection$raw_connection)
+  query <- post_process(rethinker::r()$db(DEFAULT_DB)$table(collection_name))
+  cursor <- query$run(connection$raw_connection)
```

Comment:
This is not the only place where post_process is needed. We need to add that also for code below run when changes are made to the table.